modules/Microsoft.SignalRService/webPubSub/deploy.bicep

```
@@ -0,0 +1,188 @@
+param disableLocalAuth bool = true
+
+@description('Optional. Control permission for data plane traffic coming from public networks while private endpoint is enabled. For security reasons, this value should be set to `Disabled`.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = 'Enabled'
+
+@allowed([
+  'ConnectivityLogs'
+  'MessagingLogs'
+])
+@description('Optional. Control permission for data plane traffic coming from public networks while private endpoint is enabled.')
+param resourceLogConfigurationsToEnable array = [
+  'ConnectivityLogs'
+  'MessagingLogs'
+]
+
+@description('Optional. Request client certificate during TLS handshake if enabled.')
+param clientCertEnabled bool = false
+
+@description('Optional. Networks ACLs, this value contains IPs to whitelist and/or Subnet information. For security reasons, it is recommended to set the DefaultAction Deny.')
+param networkAcls array = []
+
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+var resourceLogConfiguration = [for configuration in resourceLogConfigurationsToEnable: {
+  name: configuration
+  enabled: 'true'
+}]
+
+var identityType = systemAssignedIdentity ? 'SystemAssigned' : !empty(userAssignedIdentities) ? 'UserAssigned' : 'None'
+
+var identity = {
+  type: identityType
+  userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : null
+}
+
+@description('Optional. Network ACLs for the resource. The values for the \'allow\' and \'deny\' array can be one or more of: ClientConnection, ServerConnection, RESTAPI.')
+var webPubSubNetworkAcls = [for acl in networkAcls: {
+  defaultAction: !empty(acl.defaultAction) ? contains([ 'Allow', 'Deny' ], acl.defaultAction) ? acl.defaultAction : 'Deny' : null
+  publicNetwork: {
+    allow: !empty(acl.allow) ? acl.allow : []
+    deny: !empty(acl.deny) ? acl.deny : []
+  }
+}]
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource webPubSub 'Microsoft.SignalRService/webPubSub@2021-10-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    capacity: capacity
+    name: sku
+    tier: sku == 'Standard_S1' ? 'Standard' : 'Free'
+  }
+  identity: identity
+  properties: {
+    disableAadAuth: disableAadAuth
+    disableLocalAuth: disableLocalAuth
+    networkACLs: !empty(webPubSubNetworkAcls) && sku != 'Free_F1' ? any(webPubSubNetworkAcls) : null
+    publicNetworkAccess: publicNetworkAccess
```

Same proposal done for Redis Cache review
```suggestion
    publicNetworkAccess: !empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(privateEndpoints) ? 'Disabled' : null)
```
```suggestion
    publicNetworkAccess: publicNetworkAccess
```